### PR TITLE
T11489

### PR DESCRIPTION
**Pull request by wgreenberg**
http://phabricator.endlessm.com/T11489

### CODE REVIEW

File: src/eos-shard-dictionary.c [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+}
+
+/* Find the block for a given key with a binary search. */
+static gboolean
+dictionary_find_block (EosShardDictionary *dictionary, char *key,
+                  struct dictionary_block_table_entry *block_out)
+{
+  int fd = dictionary->fd;
+
+  int key_size = CSTRING_SIZE(key);
```
Comment (magcius) [inline src/eos-shard-dictionary.c]: nit: space before (

File: src/eos-shard-dictionary.c [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  if (memcmp (header->magic, DICTIONARY_MAGIC, sizeof (header->magic)) != 0)
+    return FALSE;
+
+  return TRUE;
+}
+
+/* Find the block for a given key with a binary search. */
+static gboolean
+dictionary_find_block (EosShardDictionary *dictionary, char *key,
+                  struct dictionary_block_table_entry *block_out)
```
Comment (magcius) [inline src/eos-shard-dictionary.c]: nit: please align

File: src/eos-shard-types.h [hunk 3]
```diff
@@ -23,3 +23,5 @@ typedef struct _EosShardShardFile EosShardShardFile;
 typedef struct _EosShardRecord EosShardRecord;
 typedef struct _EosShardBlob EosShardBlob;
 typedef struct _EosShardBlobStream EosShardBlobStream;
+typedef struct _EosShardDictionary EosShardDictionary;
+typedef struct _EosShardDictionaryWriter EosShardDictionaryWriter;
```
Comment (magcius) [inline src/eos-shard-types.h]: Would like to see tests for `EosShardDictionary` / `EosShardDictionaryWriter`.

Comment (wgreenberg) [inline src/eos-shard-types.h]: Added tests

**Pull request merged**